src/packages/auth/functions.ts

```diff
@@ -129,11 +129,21 @@ export function decodeJwt(authorizationHeader: string | undefined): JwtDecode {
   }
 }
 
-export function generateJwt(username: string, defaultRole: string, allowedRoles: string[]): string | null {
+export function generateJwt(
+  username: string,
+  defaultRole: string,
+  allowedRoles: string[],
+  expiry?: string,
+): string | null {
+  if (typeof expiry === 'undefined') {
+    const { JWT_EXPIRATION } = getEnv();
+    expiry = JWT_EXPIRATION;
+  }
+
   try {
-    const { HASURA_GRAPHQL_JWT_SECRET, JWT_EXPIRATION } = getEnv();
+    const { HASURA_GRAPHQL_JWT_SECRET } = getEnv();
     const { key, type }: JwtSecret = JSON.parse(HASURA_GRAPHQL_JWT_SECRET);
-    const options: jwt.SignOptions = { algorithm: type as Algorithm, expiresIn: JWT_EXPIRATION };
+    const options: jwt.SignOptions = { algorithm: type as Algorithm, expiresIn: expiry };
```

Comment:
...or by using a nullish coalescing operator:
```suggestion
    const options: jwt.SignOptions = { algorithm: type as Algorithm, expiresIn: expiry ?? JWT_EXPIRATION };
```

Reply:
Resolving as I took your other suggestion.